core/eolearn/core/eodata_io.py

```diff
@@ -376,23 +376,24 @@ def __init__(self, path: str, filesystem: FS):
         """
         :param path: A path in the filesystem
         :param filesystem: A filesystem object
-        :compress_level: The compression level to be used when saving, inferred from path if not provided
         """
-        filename = fs.path.basename(path)
-        expected_extension = f".{self.get_file_format().extension}"
-        compressed_extension = expected_extension + f".{MimeType.GZIP.extension}"
-        if not filename.endswith((expected_extension, compressed_extension)):
-            raise ValueError(f"FeatureIO expects a filepath with the {expected_extension} file extension, got {path}")
+        self._check_path_has_valid_extension(path)
 
         self.path = path
         self.filesystem = filesystem
 
         self.loaded_value: T | None = None
 
+    def _check_path_has_valid_extension(self, path: str) -> None:
```

Comment:
Feels like this could perhaps just be `_check_path_extension`, but dunno

Reply:
good call, adjusted